### DISCUSSION TITLE
Pool Heat Pump

### DISCUSSION THREAD

**Discussion by MasterOfPuppets111** [arendst/Tasmota#12337] [Q&A] [answered]
Hello, I bought the following heat pump for my Pool. You can control it with the Tuya/Smart Life App. 
It has an ESP8266 in it, is there a way to control it if I change the firmware to tasmota?

https://www.poolex.fr/en/produit/2020-swimming-pool-heat-pump-poolex-silverline-fi

Best regards
Pierre

**Comment by barbudor** [comment, marked as answer]
May be
As it isn't listed on the template site, you may be the 1st to try.
Tuya implementation may be very manufacturer dependant especialy for non standard devices (I mean outside of lamps, switches etc that are 90% of tuya units)

Personnaly I would
Open the device to be sure you can serial flash (and serial reflash the original fw if needed)
Inf possible investigate the Esp/MCU protocol details with original fw to get confidence that Tasmota can support the details

At least without being sure you can save the original flash content and revert to the original firmware, I would not try to avoid the risk of getting a bricked expensive heat pump
